### PR TITLE
[dxgi] fix DxgiOutput::FindClosestMatchingMode() implementation

### PR DESCRIPTION
**Pull request by Mixaill**
* remove useless checks
* fixed refresh rate matching
* fixed unspecified refresh rate detection

### CODE REVIEW

File: src/dxgi/dxgi_output.cpp [hunk 1]
```diff
@@ -93,21 +93,42 @@ namespace dxvk {
...
 
+    //filter out modes with different scaling if it was set
+    if (modeToMatch.Scaling != DXGI_MODE_SCALING_UNSPECIFIED) {
+      if (modeToMatch.Scaling != DXGI_MODE_SCALING_CENTERED) {
+        Logger::warn("DxgiOutput::FindClosestMatchingMode: DXVK currently supports only DXGI_MODE_SCALING_CENTERED. Fall back to it.");
+        modeToMatch.Scaling = DXGI_MODE_SCALING_CENTERED;
+      }
+
+      for (auto it = modes.begin(); it != modes.end();) {
+         if (it->Scaling != modeToMatch.Scaling)
```
Comment (doitsujin) [inline src/dxgi/dxgi_output.cpp]: Why is this needed? Can't we just *ignore* the Scaling? If we're falling back to CENTERED anyway, I really don't see the point in filtering out any modes based on the scaling.

File: src/dxgi/dxgi_output.cpp [hunk 2]
```diff
@@ -93,21 +93,42 @@ namespace dxvk {
...
+
     //filter out modes with different scanline ordering if it was set
     if (modeToMatch.ScanlineOrdering != DXGI_MODE_SCANLINE_ORDER_UNSPECIFIED) {
+      if (modeToMatch.ScanlineOrdering != DXGI_MODE_SCANLINE_ORDER_PROGRESSIVE) {
+        Logger::warn("DxgiOutput::FindClosestMatchingMode: DXVK currently supports only DXGI_MODE_SCANLINE_ORDER_PROGRESSIVE. Fall back to it.");
+        modeToMatch.ScanlineOrdering = DXGI_MODE_SCANLINE_ORDER_PROGRESSIVE;
+      }
+
       for (auto it = modes.begin(); it != modes.end();) {
         if (it->ScanlineOrdering != modeToMatch.ScanlineOrdering)
```
Comment (doitsujin) [inline src/dxgi/dxgi_output.cpp]: Same thing here, I don't think we need this (or the clunky log message) until DXVK actually reports the scaling modes and scanline modes properly.

File: src/dxgi/dxgi_output.cpp (hunk 1 above)
Comment (Mixaill) [reply src/dxgi/dxgi_output.cpp]: sure, removed

File: src/dxgi/dxgi_output.cpp (hunk 2 above)
Comment (Mixaill) [reply src/dxgi/dxgi_output.cpp]: same, removed
